### PR TITLE
#107: add link to PR in issues (closes #107)

### LINKED ISSUE #107: add link to PR in issues
finding PRs starting from issues is painful, we do not have a way to see if there is a PR linked to the issue from the issue view.
Currently we only have the `WIP` and `in review` labels in the issue - no link and no joy makes Jack a dull boy

discussed offline with @dhinus and @gabro 

@FrancescoCioria any thoughts?

## requirements
in the issue screen:
- add a "Open PR" button in the right menu to open PRs

## specs
{optional: describe technical specs to implement this feature, if not obvious}

## misc
a **very** clear drawing of the desired output

image

### PR DESCRIPTION
**Pull request by francescogior**
Closes #107

### CODE REVIEW

File: src/plugin/addPRLinkOnIssuePage.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import $ from 'jquery';
+import includes from 'lodash/includes';
+import find from 'lodash/find';
+import last from 'lodash/last';
+
+export default function ({ issueNumber }) {
+  const toSearch = `closes #${issueNumber}`;
+  const PRLink = (find([ ...$('.discussion-item-ref-title a')], x => includes(x.text(), toSearch)) || { attr: () => {} }).attr('href');
+  if (PRLink && $('.gwfp-pr-link').length === 0) {
+    appendPRLink({ PRLink });
```
Comment (FrancescoCioria) [inline src/plugin/addPRLinkOnIssuePage.js]: why isn't `eslint` complaining here? you're calling `appendPRLink` before it's defined

**Comment by francescogior** [comment]
@FrancescoCioria what about recycling the style of the issue description preview/link.

image

This way makes easy to switch quicky between issue and pr

File: src/plugin/addPRLinkOnIssuePage.js (hunk 1 above)
Comment (francescogior) [inline src/plugin/addPRLinkOnIssuePage.js]: ¯\_(ツ)_/¯

**Comment by FrancescoCioria** [comment]
it sounds functional, but definitely not the prettiest thing ever :P let's hear what other people think so to avoid having to fix it tomorrow :)

@buildo/everyone any opinion? is it ok for you to add that bar to every issue that has an associated PR?

PS: You have today to give us feedback, otherwise this evening we'll merge it!

**Comment by giogonzo** [comment]
:+1: as long as it isn't blocking the UI (like the new issue "buttons" are, forcing to reload the page on when something goes wrong)

**Comment by gabro** [comment]
My eyes 😰. How about a button in the sidebar?

image

**Comment by FrancescoCioria** [comment]
@gabro that's current state of the PR, we're here cause actually I asked not to put it in the sidebar as there are too many buttons 😄 
If we like the sidebar ok, but IMO there are really too many things there

**Comment by danielegallingani** [comment]
Not super nice but it catches your eyes and it works, I would merge this :)

**Comment by FrancescoCioria** [comment]
ok @francescogior I'd say we leave it in the sidebar then :)

do you think you can fix that defect where the position of PR and new-sub-issue is random? (and force somehow PR button to be always the first one?)

**Comment by francescogior** [comment]
Are you sure people reviewed the button in the sidebar instead of the same ui of the issue description shown here https://github.com/buildo/github-workflow-pal/pull/117#issuecomment-282662107, (that is the current state with the last commit btw)?

Since two independent plugins use `$('same-thing').prepend()` there should be an ugly custom logic in both of them

**Comment by giogonzo** [comment]
@francescogior pics or it didn't happen

**Comment by francescogior** [comment]
gif and it did happen

**Comment by francescogior** [comment]
I will merge this PR in the current state, today at 14.35pm if @FrancescoCioria approves and nobody else complains (note: a bug was fixed yesterday and the UI improved).
The gif in the description represent the current UX

**Pull request merged**